cpp/src/barretenberg/stdlib/primitives/logic/logic.test.cpp

```diff
@@ -1,13 +1,25 @@
 #include "../bool/bool.hpp"
+#include "barretenberg/numeric/uint256/uint256.hpp"
+#include "barretenberg/proof_system/types/composer_type.hpp"
 #include "logic.hpp"
-#include "barretenberg/plonk/proof_system/constants.hpp"
 #include <gtest/gtest.h>
 #include "barretenberg/honk/composer/standard_honk_composer.hpp"
 #include "barretenberg/plonk/composer/standard_composer.hpp"
 #include "barretenberg/plonk/composer/ultra_composer.hpp"
 #include "barretenberg/plonk/composer/turbo_composer.hpp"
 #include "barretenberg/numeric/random/engine.hpp"
 
+#pragma GCC diagnostic ignored "-Wimplicitly-unsigned-literal"
```

Comment:
I suggested ignoring the unused local typedefs warning because it gave a systematic simplification and there is no risk when creating aliases that are not used, but in general we should remove these sorts of warnings before merging. this one in particular is breaking gcc compilation, so it'll have to go.